### ISSUE TITLE
Login - emails are case sensitive

### ISSUE THREAD

**Issue by tristan-orourke** [GCTC-NTGC/TalentCloud#470]
Resolved by #749, after scrutiny. Requires removing dupes on the live site ahead of deployment.

# Description
Emails are saved as case-sensitive. You must therefore use the same to log in as when you registered. However, they should be case-insensitive.

# Steps to Reproduce
1. register using an all lowercase email
2. attempt to log in using the same email but with uppercase letters.

**Comment by Khristophor** [comment]
@tristan-orourke after doing some reading and attempting some suggested solutions, it looks like the root cause of this is on the DB side. Postgres is case sensitive by default, while MySQL is not (leading to a lack of issues on this subject). Postgres has an option for this called CITEXT collation, but we'd have to install/enable/change the column collation for the `email` field on the `users` table.

https://www.postgresql.org/docs/9.2/citext.html

**Comment by Khristophor** [comment]
A (very robust) solution to adding this via a migration:
https://stackoverflow.com/a/30447403

**Comment by gobyrne** [comment]
@tristan-orourke @Khristophor My understanding is that this is on hold until we have a good place to test it out? I'll pull it out of "in progress" for now, but if it's actively being worked on, go ahead.

**Comment by gobyrne** [comment]
Dev server is now up and running, we should no longer be waiting on a dependency.

**Comment by xander-miller** [comment]
Couldn't we just normalize all the emails to downcase in the controller before it even gets to the database? We could then run a migration to downcase all the emails already in the database.
